src/util/timer.h

```diff
@@ -58,16 +65,15 @@ class ScopedTimer {
 #else
         QString strKey = arg.isEmpty() ? key.toString() : key.toString().arg(arg);
 #endif
-        m_state = std::make_optional<TimerData>(TimerData{Timer(strKey, compute), false});
-        m_state->timer.start();
+        m_maybeTimer = std::make_optional<Timer>(std::move(strKey), compute);
+        m_maybeTimer->start();
     }
 
     ~ScopedTimer() noexcept {
-        if (m_state) {
-            if (!m_state->cancelled) {
-                m_state->timer.elapsed(true);
-            }
+        if (m_maybeTimer) {
+            m_maybeTimer->elapsed(true);
         }
+        m_maybeTimer.reset();
     }
```

Comment:
The reset triggers a `-Wmaybe-uninitialized` warning locally for me and I'm not sure why...

```cpp

In destructor ‘QString::~QString()’,
    inlined from ‘Timer::~Timer()’ at /home/swiftb0y/Sourcerepositories/mixxx/src/util/timer.h:20:7,
    inlined from ‘constexpr void std::_Optional_payload_base<_Tp>::_M_destroy() [with _Tp = Timer]’ at /usr/include/c++/13/optional:287:35,
    inlined from ‘constexpr void std::_Optional_payload_base<_Tp>::_M_reset() [with _Tp = Timer]’ at /usr/include/c++/13/optional:318:14,
    inlined from ‘constexpr std::_Optional_payload<_Tp, false, _Copy, _Move>::~_Optional_payload() [with _Tp = Timer; bool _Copy = false; bool _Move = false]’ at /usr/include/c++/13/optional:439:65,
    inlined from ‘constexpr std::_Optional_base<Timer, false, false>::~_Optional_base()’ at /usr/include/c++/13/optional:510:12,
    inlined from ‘constexpr std::optional<Timer>::~optional()’ at /usr/include/c++/13/optional:705:11,
    inlined from ‘ScopedTimer::~ScopedTimer()’ at /home/swiftb0y/Sourcerepositories/mixxx/src/util/timer.h:71:5,
    inlined from ‘virtual void VinylControlXwax::analyzeSamples(CSAMPLE*, size_t)’ at /home/swiftb0y/Sourcerepositories/mixxx/src/vinylcontrol/vinylcontrolxwax.cpp:648:1:
/usr/include/qt5/QtCore/qstring.h:1311:35: warning: ‘*(QString*)((char*)&t + offsetof(ScopedTimer, ScopedTimer::m_maybeTimer.std::optional<Timer>::<unnamed>.std::_Optional_base<Timer, false, false>::<unnamed>)).QString::d’ may be used uninitialized [-Wmaybe-uninitialized]
 1311 | inline QString::~QString() { if (!d->ref.deref()) Data::deallocate(d); }

```

Reply:
The Uninitialized QString seems to be Timer::m_key. The warning happens after the user destructor in the auto generated code. 
It is uninitialized if m_maybeTimer is std::nullopt. But in that case the ~Timer() not be run.